defi/src/storeTvlInterval/storeNewTvl.ts

```diff
@@ -144,41 +147,52 @@ export default async function (
     }
   }
 
-  let tvlPrev1Day =  (await lastDailyTVLRecord).tvl
+  let tvlPrev1Day = (await lastDailyTVLRecord).tvl
   let tvlPrev1Week = (await lastWeeklyTVLRecord).tvl
   const dayDailyTvl = (await dayDailyTvlRecord).tvl
   const weekDailyTvl = (await weekDailyTvlRecord).tvl
-  if(tvlPrev1Day !== 0 && dayDailyTvl !== 0 && tvlPrev1Day > (dayDailyTvl*2)){
+  if (tvlPrev1Day !== 0 && dayDailyTvl !== 0 && tvlPrev1Day > (dayDailyTvl * 2)) {
     tvlPrev1Day = 0;
   }
-  if(tvlPrev1Week !== 0 && weekDailyTvl !== 0 && tvlPrev1Week > (weekDailyTvl*2)){
+  if (tvlPrev1Week !== 0 && weekDailyTvl !== 0 && tvlPrev1Week > (weekDailyTvl * 2)) {
     tvlPrev1Week = 0;
   }
 
-  await dynamodb.put({
-    PK: hourlyPK,
-    SK: unixTimestamp,
+  const hourlyData = {
     ...tvl,
     ...(storePreviousData
       ? {
-          tvlPrev1Hour: lastHourlyTVLObject.tvl,
-          tvlPrev1Day,
-          tvlPrev1Week,
-        }
-      : {}),
+        tvlPrev1Hour: lastHourlyTVLObject.tvl,
+        tvlPrev1Day,
+        tvlPrev1Week,
+      }
+      : {})
+  }
+  await dynamodb.put({
+    PK: hourlyPK,
+    SK: unixTimestamp,
+    ...hourlyData,
   });
 
-  const closestDailyRecord = await getTVLOfRecordClosestToTimestamp(
+  const dayTimestamp = getTimestampAtStartOfDay(unixTimestamp);
+  const writeOptions = { overwriteExistingData };
+  await Promise.all([
+    saveProtocolItem(hourlyTvl, { id: protocol.id, timestamp: unixTimestamp, data: hourlyData, }, writeOptions),
+    saveProtocolItem(dailyTvl, { id: protocol.id, timestamp: unixTimestamp, data: tvl, }, writeOptions),
```

Comment:
shouldnt timestamp here be dayTimestamp instead?

Reply:
yes, this is a bug